integration_tests/test_group.py

```diff
@@ -0,0 +1,215 @@
+    )
+
+    # submit a proposal
+    proposal_file = tmp_path / "proposal.json"
+    proposal = {
+        "group_policy_address": group_policy_address,
+        "messages": [
+            {
+                "@type": "/cosmos.bank.v1beta1.MsgSend",
+                "from_address": group_policy_address,
+                "to_address": signer1,
+                "amount": [{"denom": "basecro", "amount": "100000000"}],
+            }
+        ],
+        "metadata": "proposal",
+        "proposers": [admin],
+    }
+
+    with open(proposal_file, "w") as f:
+        json.dump(proposal, f)
+
+    rsp = json.loads(
+        cli.raw(
+            "tx",
+            "group",
+            "submit-proposal",
+            proposal_file,
+            "-y",
+            _from="community",
+            home=cli.data_dir,
+            node=cli.node_rpc,
+            keyring_backend="test",
+            chain_id=cli.chain_id,
+        )
+    )
+
+    assert rsp["code"] == 0, rsp["raw_log"]
+
+    proposal_id = json.loads(
+        json.loads(rsp["raw_log"])[0]["events"][0]["attributes"][0]["value"]
+    )
+
+    # vote on proposal
+    rsp = json.loads(
+        cli.raw(
+            "tx",
+            "group",
+            "vote",
+            proposal_id,
+            signer1,
+            "VOTE_OPTION_YES",
+            "vote-signer1-yes",
+            "-y",
+            _from="signer1",
+            home=cli.data_dir,
+            node=cli.node_rpc,
+            keyring_backend="test",
+            chain_id=cli.chain_id,
+        )
+    )
+
+    assert rsp["code"] == 0, rsp["raw_log"]
+
+    rsp = json.loads(
+        cli.raw(
+            "tx",
+            "group",
+            "vote",
+            proposal_id,
+            signer2,
+            "VOTE_OPTION_YES",
+            "vote-signer2-yes",
+            "-y",
+            _from="signer2",
+            home=cli.data_dir,
+            node=cli.node_rpc,
+            keyring_backend="test",
+            chain_id=cli.chain_id,
+        )
+    )
+
+    assert rsp["code"] == 0, rsp["raw_log"]
+
+    # query proposal votes
+    rsp = json.loads(
+        cli.raw(
+            "query",
+            "group",
+            "votes-by-proposal",
+            proposal_id,
+            home=cli.data_dir,
+            node=cli.node_rpc,
+            chain_id=cli.chain_id,
+        )
+    )
+
+    assert len(rsp["votes"]) == 2, rsp
+
+    # transfer some amount to group policy address
+    cluster.transfer(admin, group_policy_address, "1cro")
+
+    # execute proposal
+    rsp = json.loads(
+        cli.raw(
+            "tx",
+            "group",
+            "exec",
+            proposal_id,
+            "-y",
+            _from="community",
+            home=cli.data_dir,
+            node=cli.node_rpc,
+            keyring_backend="test",
+            chain_id=cli.chain_id,
+        )
+    )
+
+    assert rsp["code"] == 0, rsp["raw_log"]
+
+    # check if the proposal executed successfully
```

Comment:
two more assertion to check that the amount was transferred successfully (deducted from the group / sender, and added to the receiver)?

Reply:
Done